### PR TITLE
feat: error reporting

### PR DESCRIPTION
**Pull request by alinauroz**

### CODE REVIEW

File: ui/utils/reportError.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const fd = new FormData();
+    fd.append("content", "APP CRASH NOTIFICATION");
+    fd.append(
+      "file",
+      new Blob([content], { type: "text/plain" }),
+      "log-" + Date.now() + ".json"
+    );
+    fd.append(
+      "avatar_url",
+      "https://media3.giphy.com/media/cLqxgg4nke0iu8UpzD/giphy.gif"
```
Comment (Powersource) [inline ui/utils/reportError.ts:21]: if we're allowed to use this image then download it and use directly instead

File: ui/.env.local.default [hunk 2]
```diff
@@ -24,3 +24,4 @@ FROM_EMAIL="fake@example.com"
 #STRIPE_YEARLY_PLAN_PRICE_ID
 
 #TERMS_UPDATED_AT="2022-05-30T20:18:35.225Z"
+#ERROR_REPORTING_WEBHOOK=
```
Comment (Powersource) [inline ui/.env.local.default:27]: what should we set this to in prod?

**Comment by Powersource** [comment]
@alinauroz which issue is this for?

**Comment by aerugo** [comment]
> @alinauroz which issue is this for?

This has not been tracked in a corresponding issue. @alinauroz suggested working on this in a weekly tech meeting and we agreed.

File: ui/.env.local.default (hunk 2 above)
Comment (alinauroz) [reply ui/.env.local.default:27]: A new webhook should be created on discord. In case of no webhook, there will be no error message in discord

**Comment by aerugo** [comment]
@alinauroz conflicts here too

File: ui/.env.local.default (hunk 2 above)
Comment (Powersource) [reply ui/.env.local.default:27]: Why on discord? Wouldn't that very quickly become unmanageable? But I guess we could maybe try it as a first iteration

Comment (Powersource) [reply ui/.env.local.default:27]: added now

**Comment by Powersource** [comment]
Added localizations for the new strings but ran into this https://github.com/cobudget/cobudget/issues/851

File: ui/utils/reportError.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    const fd = new FormData();
+    fd.append("content", "APP CRASH NOTIFICATION");
+    fd.append(
+      "file",
+      new Blob([content], { type: "text/plain" }),
+      "log-" + Date.now() + ".json"
+    );
+    fd.append("avatar_url", "https://cobudget.com/favicon.ico");
+
+    fetch(hookUrl, {
```
Comment (Powersource) [inline ui/utils/reportError.ts:21]: @alinauroz I get these errors in my browser console when I cause this to run

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://discord.com/api/webhooks/1025000522022920192/1xjiaJ86fl2EzhJoUOxNwOdW1kIeRM2Krgql1fKlmUH9WJ62zAt11QN4xdEoU93N_jzW. (Reason: CORS request did not succeed). Status code: (null).

Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.
```

File: ui/utils/reportError.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+export default function reportError(error: Error) {
+  const { message, stack } = error;
+  const hookUrl = process.env.ERROR_REPORTING_WEBHOOK;
+  if (hookUrl) {
+    const content = JSON.stringify({
+      error,
+      message,
+      stack,
+      url: window?.location,
+    });
```
Comment (Powersource) [inline ui/utils/reportError.ts:10]: If we do `JSON.stringify(..., nulll, 2)` then it'll be easier to read on discord

File: ui/utils/reportError.ts (hunk 3 above)
Comment (Powersource) [reply ui/utils/reportError.ts:24]: simply removed it

**Pull request merged**